src/workerd/api/js-rpc-service.c++

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "js-rpc-service.h"
+
+namespace workerd::api {
+
+GetJsRpcTargetEvent::GetJsRpcTargetEvent()
+    : ExtendableEvent("getJsRpcTarget") {};
+
+kj::Promise<WorkerInterface::CustomEvent::Result> GetJsRpcTargetCustomEventImpl::run(
+    kj::Own<IoContext::IncomingRequest> incomingRequest,
+    kj::Maybe<kj::StringPtr> entrypointName) {
+  incomingRequest->delivered();
+  auto targetPaf = kj::newPromiseAndFulfiller<void>();
+  auto callPromise = kj::mv(targetPaf.promise);
+  capFulfiller->fulfill(kj::heap<JsRpcTargetImpl>(
+      kj::mv(targetPaf.fulfiller), incomingRequest->getContext(), entrypointName));
+
+  incomingRequest->getContext().addWaitUntil(callPromise.attach(
+      kj::defer([incomingRequest=kj::mv(incomingRequest)]() mutable {
+    // We can drain once our RPC call completes.
+    auto& context = incomingRequest->getContext();
+    context.addWaitUntil(incomingRequest->drain().attach(kj::mv(incomingRequest)));
+  })));
+
+  co_return WorkerInterface::CustomEvent::Result {
```

Comment:
I'd argue this should wait for the RPC session to be done (`callPromise` resolves) before returning, though I suppose there's room for debate.

I think all other CustomEvent implementations today actually wait for all waitUntils to finish?